Sources/UIGestureRecognizer.swift

```diff
@@ -29,17 +29,19 @@ open class UIGestureRecognizer {
 
     public var state: UIGestureRecognizerState = .possible {
         didSet {
-            if state == oldValue { return }
+            if
+                state == oldValue || oldValue == .cancelled && state != .ended
+            {
+                return
+            }
+
             onStateChanged?()
```

Comment:
why do we return early before calling `onStateChanged`? A code comment would be useful